wicket-core/src/main/java/org/apache/wicket/protocol/http/CsrfPreventionRequestCycleListener.java

```diff
@@ -350,15 +365,29 @@ public void onRequestHandlerResolved(RequestCycle cycle, IRequestHandler handler
 			// Check if the page should be CSRF protected
 			if (isChecked(targetedPage))
 			{
-				// if so check the Origin HTTP header
-				checkRequest(containerRequest, sourceUri, targetedPage);
+				// check sec-fetch-site header and call the fetch metadata check if present
+				if (hasFetchMetadataHeaders(containerRequest))
+				{
+					// set vary headers to avoid caching responses processed by Fetch Metadata
+					// caching these responses may return 403 responses to legitimate requests
+					// or defeat the protection
+					WebResponse webResponse = (WebResponse)cycle.getResponse();
+					webResponse.setHeader(VARY_HEADER, SEC_FETCH_DEST_HEADER + ", "
+						+ SEC_FETCH_SITE_HEADER + ", " + SEC_FETCH_MODE_HEADER);
```

Comment:
We should try to be as little intrusive as possible. Let's only set the Vary header if we act upon the current request, so let's move this inside checkRequestFetchMetadata, after checking if this host is in our allowlist (only then we will make a judgement based on FM)